baseplate/lib/metrics.py

```diff
@@ -302,6 +304,9 @@ def send(self, elapsed: float) -> None:
 
         """
         serialized = self.name + (f":{(elapsed * 1000.0):g}|ms".encode())
```

Comment:
like the counters, all the parts should be in the list building setup, not just the sample rate.

```python
        parts = [f"{self.name}:{(elapsed * 1000.0):g}".encode()), b"ms"]

        if sample_rate < 1.0:
            parts.append(f"@{sample_rate:g}".encode())

        serialized = b"|".join(parts)
        self.transport.send(serialized)
```

Reply:
This gets complicated because name is bytes but everything else is strings, so name gets double-encoded.